codegenerator/cli/src/rescript_types.rs

```diff
@@ -744,14 +754,14 @@ mod tests {
                 RescriptRecordField::new("fieldA".to_string(), RescriptTypeIdent::Int),
                 RescriptRecordField::new("fieldB".to_string(), RescriptTypeIdent::Bool),
             ])
-            .to_rescript_schema(),
-            "S.object(s => {fieldA: s.field(\"fieldA\", S.int), fieldB: s.field(\"fieldB\", \
+            .to_rescript_schema(&"eventArgs".to_string()),
+            "S.object((s): eventArgs => {fieldA: s.field(\"fieldA\", S.int), fieldB: s.field(\"fieldB\", \
```

Comment:
What is the error you get if you don't have the type annotation? Does rescript not infer the type if it's in the same scope?

Reply:
There are cases when there are multiple types with the same shape, and ReScript doesn't know which one to use